specification/communication/data-plane/NetworkTraversal/preview/2021-09-15-preview/CommunicationNetworkTraversal.json

```diff
@@ -0,0 +1,169 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "CommunicationNetworkingClient",
+    "description": "Azure Communication Networking Service",
+    "version": "2021-09-15-preview"
+  },
+  "paths": {
+    "/networktraversal/:issueRelayConfiguration": {
+      "post": {
+        "tags": [
+          "Turn"
+        ],
+        "summary": "Issue a configuration for an STUN/TURN server for an existing identity.",
+        "operationId": "CommunicationNetworkTraversal_IssueRelayConfiguration",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CommunicationRelayConfigurationRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/CommunicationRelayConfiguration"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Issue Relay Configuration": {
+            "$ref": "./examples/IssueRelayConfiguration.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CommunicationRelayConfigurationRequest": {
+      "description": "Request for a CommunicationRelayConfiguration.",
+      "required": [
+        "routeType"
```

Comment:
@DominikMe  Should the routeType be required? I see in the design doc there's set that a default value for it, which makes me think it's not necessary to make it rquired. 

Also, if both userId and routeType are not required in the request. Should the request not be required then? Or is it better to have the request as required in case we add a required field later.

Reply:
Cc. @ajpeacock0

Reply:
It depends if you want the client to be explicitly aware which routeType they request. If not it doesn't need to be required.

I think the whole request body can be `"required": false` if all properties are optional but I would double check what code AutoRest then generates for each language.

Reply:
My opinion on `routeType` from the background in the doc is that it can be optional and developers don't need to send it in the request.

That's different from issuing an Acs user token, where we decided to make `scopes` mandatory because we felt hat a default scope is more confusing than helpful.

Reply:
I ran autorest with this change and it shows that it doesn't require the user ID anymore (as expected). I noticed there's no difference in the autogenerated code for the Request (CommunicationRelayConfigurationRequest class) and in the implementations of the client whether the request is required or not. 

Anything I should check in particular from the autogenerated code?

Reply:
That sounds reasonable. You can run the live tests if you have the service version deployed.